test/extended/cli/explain.go

```diff
@@ -620,9 +620,12 @@ var _ = g.Describe("[sig-cli] oc explain networking types", func() {
 })
 
 func verifySpecStatusExplain(oc *exutil.CLI, crdClient apiextensionsclientset.Interface, gvr schema.GroupVersionResource) error {
-	return verifyExplain(oc, crdClient, gvr,
-		`(?s)DESCRIPTION:.*FIELDS:.*spec.*<Object>.*[Ss]pec(ification)?.*status.*<Object>.*[Ss]tatus.*`,
-		gvr.Resource, fmt.Sprintf("--api-version=%s", gvr.GroupVersion()))
+	singularResourceName, _ := strings.CutSuffix(gvr.Resource, "s")
```

Comment:
ye, it is just a quick solution to make it work for the types we test (`builtinTypes`). It will not work for the special cases like `ingresses`, etc. 

- We could fix this later once we add such a special type in the future. 
- Or I can add a prerequisite that a singular should be specified for each resource in the test in builtinTypes. 
- Or create a singularExceptions map for a few of these resources. Thoughts?

Reply:
After thinking on this; maybe it is not necessary try to add singular name of resource between `<` and `>`. It seems to me that the maintenance effort outweighs the value we get comparing to the simply `<.*>`

Reply:
Up to you, but I think it is safe to leave it since we are probably not expecting new builtin types. And if there are new in the future, exception can be made by the person introducing these.

Reply:
I agree that it would be better being at safe side but I'd still prefer not stealing someone's at least 4 hours in the future by waiting for the CI and after failing, adding an exception into this regex :). Let's defer final decision to @soltysh as an approver.